core/chains/evm/client/rpc_client.go

```diff
@@ -1349,30 +1284,10 @@ func (r *RPCClient) acquireQueryCtx(parentCtx context.Context, timeout time.Dura
 		cp := *r.http
 		http = &cp
 	}
-	r.stateMu.RUnlock()
-	ctx, cancel = makeQueryCtx(parentCtx, chStopInFlight, timeout)
+	ctx, cancel = multinode.MakeQueryCtx(parentCtx, chStopInFlight, timeout)
```

Comment:
Do we need to `MakeQueryCtx` can Adapter implement single method  `func (m *Adapter[RPC, HEAD]) AcquireQueryCtx(parentCtx context.Context, timeout time.Duration) (ctx context.Context, cancel context.CancelFunc, chStopInFlight chan) struct{}`.
That does not return rpc (as it's not needed in the adapter)

Reply:
Good point, we can certainly simplify this!